soap/index.d.ts

```diff
@@ -19,12 +19,22 @@ export class ClientSSLSecurity implements Security {
     constructor(key: Buffer | string, cert: Buffer | string, defaults?: any);
 }
 
-interface Client extends events.EventEmitter {
-    setSecurity(s: Security): void;
-    [method: string]: (args: any, fn: (err: any, result: any) => void, options?: any, extraHeaders?: any) => void;
-    addSoapHeader(headJSON: any): void;
-    setEndpoint(endpoint: string): void;
+export interface Client extends events.EventEmitter {
+    [method: string]: any;
```

Comment:
SOAP service method was a function type. You changed it to `any`. It doesn't match [document](https://github.com/vpulim/node-soap#clientmethodargs-callback---call-method-on-the-soap-service).

```
[method: string]: (args: any, fn: (err: any, result: any) => void, options?: any, extraHeaders?: any) => void;
```

I remember somebody in TypeScript team gave a comment on one of my PR. He said that definition which match all should be put at back. But I can't find out the comment just now.